File: Python/Program/OperatingSystem.py
from SoftWare import SoftWare

class OperatingSystem(SoftWare):
    def __init__(self, name="", version="", architecture="", kernel=""):
        super().__init__(name, version)
        self.architecture = architecture
        self.kernel = kernel

    def set_architecture(self, architecture):
        self.architecture = architecture

    def set_kernel(self, kernel):
        self.kernel = kernel

    def get_architecture(self):
        return self.architecture

    def get_kernel(self):
        return self.kernel

File: Java/Program/OperatingSystem.java
public class OperatingSystem extends SoftWare {
    private String architecture;
    private String kernel;

    public OperatingSystem() {}

    public OperatingSystem(String name, String version, String architecture, String kernel) {
        super(name, version); 
        this.architecture = architecture;
        this.kernel = kernel;
    }

    public void setArchitecture(String architecture) { this.architecture = architecture; }
    public void setKernel(String kernel) { this.kernel = kernel; }

    public String getArchitecture() { return architecture; }
    public String getKernel() { return kernel; }
}
